### PR TITLE
LPS-125550 log4j2 upgrade

### PR DESCRIPTION
**Pull request by yuhai**
Hi @tinatian, the pr rebased on https://github.com/shuyangzhou/liferay-portal/pull/9829.

cc @dantewang

### CODE REVIEW

**Comment by yuhai** [comment]
ci:test:sf

**Comment by yuhai** [comment]
ci:test:relevant

**Comment by tinatian** [comment]
@yuhai The folder name should be `org-apache-logging-log4j-core` instead of `org.apache.logging.log4j.core`, please resend

File: classes/log4j2.properties [hunk 1]
```diff
@@ -0,0 +1,14 @@
+appenders=console
```
Comment (tinatian) [inline classes/log4j2.properties:1]: @yuhai For this pull, we should call it "rename" as it actually changed the content of log4j.properties, please reword.

Thanks.

File: modules/apps/portal/portal-log4j-test/src/testIntegration/java/com/liferay/portal/log4j/test/PortalLog4jTest.java [hunk 2]
```diff
@@ -491,16 +504,47 @@ private void _testLogOutput(
...
 	private static Path _xmlLogFilePath;
 
+	private static class TestOutputStream extends CloseShieldOutputStream {
+
+		public TestOutputStream(OutputStream originalOutputStream) {
+			super(originalOutputStream);
+		}
+
+		@Override
+		public void write(byte[] b) throws IOException {
```
Comment (tinatian) [inline modules/apps/portal/portal-log4j-test/src/testIntegration/java/com/liferay/portal/log4j/test/PortalLog4jTest.java:528]: @yuhai Please rename `byte[] b` to `byte[] bytes`

Comment (tinatian) [reply modules/apps/portal/portal-log4j-test/src/testIntegration/java/com/liferay/portal/log4j/test/PortalLog4jTest.java:528]: @yuhai Please simplify the logic :
`String content = new String(b);
_unsyncStringWriter.write(content.toCharArray());`

**Pull request closed without merging**